Mathlib/AlgebraicGeometry/Morphisms/Separated.lean

```diff
@@ -128,6 +127,40 @@ lemma IsSeparated.of_comp [IsSeparated (f ≫ g)] : IsSeparated f := by
 lemma IsSeparated.comp_iff [IsSeparated g] : IsSeparated (f ≫ g) ↔ IsSeparated f :=
   ⟨fun _ ↦ .of_comp f g, fun _ ↦ inferInstance⟩
 
+@[stacks 01KM]
+instance isClosedImmersion_equalizer_ι_left {S : Scheme} {X Y : Over S} [IsSeparated Y.hom]
+    (f g : X ⟶ Y) : IsClosedImmersion (equalizer.ι f g).left := by
+  refine IsClosedImmersion.stableUnderBaseChange
+    ((Limits.isPullback_equalizer_prod f g).map (Over.forget _)).flip ?_
+  rw [← MorphismProperty.cancel_right_of_respectsIso @IsClosedImmersion _
+    (Over.prodLeftIsoPullback Y Y).hom]
+  convert (inferInstanceAs (IsClosedImmersion (pullback.diagonal Y.hom)))
+  ext1 <;> simp [← Over.comp_left]
+
+/--
+Suppose `X` is a reduced scheme and that `f g : X ⟶ Y` agree over some separated `Y ⟶ Z`.
+Then `f = g` if `ι ≫ f = ι ≫ g` for some dominant `ι`.
+-/
+lemma ext_of_isDominant [IsReduced X] {f g : X ⟶ Y}
```

Comment:
Should this be `ext_of_isSeparated_of_isDominant`, and then we have two 1-line special cases where `s` / `\i` is the identity?

Reply:
If `i` or `s` is the identity then the lemma is trivial I think.
But I can provide one where `s : _ -> Spec Z`.

Reply:
Of course, you are totally right. Thanks for interpreting my comment in a useful way!